### PR TITLE
Fix lr_scheduler in no_trainer training scripts

### PR DESCRIPTION
**Pull request by bofenghuang**
# What does this PR do?

Hi!

I ran into unexpected behaviors with the LR scheduler when using gradient accumulation in a distributed environment within this script. I managed to fix it on my end with the following changes:

1. `AcceleratedScheduler` currently steps only when gradient accumulation is performed. As a result, there's no need to multiply `gradient_accumulation_steps` when setting up `num_warmup_steps` and `num_training_steps` for `get_scheduler`, since it will be handled inside `AcceleratedScheduler`.

https://github.com/huggingface/accelerate/blob/6a4857fec2bbae83880014cfd834d8a3e22de68b/src/accelerate/scheduler.py#L60-L64

2. In distributed settings, the scheduler steps for `accelerator.num_processes` each time, requiring multiplying `num_warmup_steps` by `accelerator.num_processes` here to guarantee proper warm-up.

3. When handling distributed training and setting `max_train_steps` for total training steps instead of `num_train_epochs`, `num_training_steps` for the scheduler should be multiplied by `accelerator.num_processes` as well, to ensure the correct number of updating steps.

I could also update other scripts facing a similar issue, if this modification is validated :)

Thanks in advance!

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

cc @muellerzr @pacman100

### CODE REVIEW

**Comment by bofenghuang** [comment]
Kindly ping @muellerzr

**Review by muellerzr — COMMENTED** [review]
Indeed the result here makes sense to me all things considered. Please go ahead and propagate them to the rest of the examples, thanks! :)

**Comment by bofenghuang** [comment]
Hi @muellerzr, thanks for the review! Just updated other no_trainer scripts which handle gradient accumulation by `with accelerator.accumulate(model)`

**Review by muellerzr — APPROVED** [review]
Thanks!

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing!
